PXC-3641: [MTR] Upgrade tests fail due to PXC users

venkatesh-prasad-v opened the pull request:
https://jira.percona.com/browse/PXC-3641

Problem
---
Due to presence of pxc internal users mysql.pxc.internal.session and
mysql.pxc.sst.role which already had FLUSH_* privileges, the logic of
fixing system tables was broken during the upgrade of the server with
--upgrade=FORCE, and the below tests were failing.

1. ps_sys_upgrade - failed with result content mismatch due to the PXC
   user `mysql.pxc.internal.session`.

2. transactional_acl_tables - failed with result content mismatch due to
   the PXC user  `mysql.pxc.internal.session`.

3. mysql_upgrade_grant - failed due to result content mismatch because
   of the missing XA_RECOVER_ADMIN privilege of a super user after the
   upgrade. This failure is also because of the user
   `mysql.pxc.internal.session`.

Fix
---

Adjusted scripts/mysql_system_tables_fix.sql to exclude PXC internal
users while adding FLUSH_* privileges to users with RELOAD privilege
(during --upgrade=FORCE).


Testing Done
---
Full MTR: https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/284/console 
Galera & sys var: https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/285/console

Failing tests are unrelated to this patch.

satya-bodapati commented on scripts/mysql_system_tables_fix.sql:
```diff
@@ -1303,7 +1303,9 @@ DROP PREPARE stmt;
 #  GRANT BACKUP_ADMIN, LOCK TABLES, PROCESS, RELOAD, REPLICATION CLIENT, SUPER ON *.* TO 'mysql.pxc.internal.session'@localhost WITH GRANT OPTION;
 #
-INSERT IGNORE INTO mysql.user VALUES ('localhost','mysql.pxc.internal.session','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','','','','',0,0,0,0,'caching_sha2_password','$A$005$THISISACOMBINATIONOFINVALIDSALTANDPASSWORDTHATMUSTNEVERBRBEUSED','N',CURRENT_TIMESTAMP,NULL,'Y','Y','Y',NULL,NULL,NULL,NULL);
+INSERT IGNORE INTO mysql.user
+(host, user, select_priv, plugin, authentication_string, ssl_cipher, x509_issuer, x509_subject)
+VALUES ('localhost','mysql.pxc.internal.session','Y','caching_sha2_password','$A$005$THISISACOMBINATIONOFINVALIDSALTANDPASSWORDTHATMUSTNEVERBRBEUSED','','','');
```
why this INSERT statement is changed? just simplified?

venkatesh-prasad-v replied:
I had made these fixes few months back and it had fixed a few test failures. But now, I have forgotten the exact reason. I'll dig into it and will let you know.

venkatesh-prasad-v replied:
I can't recall the reason for this change. When I tried with reverting this change, it didn't cause any test failures in my local environment. So, I have submitted to Jenkins with reverted changes for further testing.

Branch: https://github.com/venkatesh-prasad-v/percona-xtradb-cluster/tree/8.0-PXC-3641-jenkins
Full suite: https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/289/console
Galera suite: https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/290/console

satya-bodapati replied:
ok, lets revert it

venkatesh-prasad-v replied:
I'll wait for a clean Jenkins and will then update the PR.

satya-bodapati replied:
:+1:

venkatesh-prasad-v replied:
Test results were good after reverting the INSERT statement changes, I have updated the PR with the required changes.

Commit:
PXC-3641: [MTR] Upgrade tests fail due to PXC users

https://jira.percona.com/browse/PXC-3641

Problem
=======
Due to presence of pxc internal users mysql.pxc.internal.session and
mysql.pxc.sst.role which already had FLUSH_* privileges, the logic of
fixing system tables was broken during the upgrade of the server with
--upgrade=FORCE, and the below tests were failing.

1. ps_sys_upgrade - failed with result content mismatch due to the PXC
   user `mysql.pxc.internal.session`.

2. transactional_acl_tables - failed with result content mismatch due to
   the PXC user  `mysql.pxc.internal.session`.

3. mysql_upgrade_grant - failed due to result content mismatch because
   of the missing XA_RECOVER_ADMIN privilege of a super user after the
   upgrade. This failure is also because of the user
   `mysql.pxc.internal.session`.

Fix
===

1. Adjusted scripts/mysql_system_tables_fix.sql to exclude PXC internal
   users while adding FLUSH_* privileges to users with RELOAD privilege
   (during --upgrade=FORCE).

2. Re-recorded the failing tests.

satya-bodapati reviewed (approved):
Please squash and merge
